src/main/java/net/minecraftforge/common/extensions/IForgeTagAppender.java

```diff
@@ -66,4 +66,82 @@
     {
         return self().add(new Tag.OptionalTagEntry(location));
     }
+    
+    /**
+     * Adds a registry entry to the tag json's remove list. Callable during datageneration.
+     * @param element The entry to remove
+     * @return The builder for chaining
+     */
+    default TagsProvider.TagAppender<T> removeRegistryEntry(final T entry)
+    {
+        return removeElementByID(this.self().registry.getKey(entry));
+    }
+    
+    /**
+     * Adds multiple registry entries to the tag json's remove list. Callable during datageneration.
+     * @param entries The entries to remove
+     * @return The builder for chaining
+     */
+    @SuppressWarnings("unchecked")
+    default TagsProvider.TagAppender<T> removeRegistryEntries(final T...entries)
+    {
+        for (T entry : entries)
+        {
+            this.removeRegistryEntry(entry);
+        }
+        return self();
+    }
+    
+    /**
+     * Adds a single element's ID to the tag json's remove list. Callable during datageneration.
+     * @param location The ID of the element to remove
+     * @return The builder for chaining
+     */
+    default TagsProvider.TagAppender<T> removeElementByID(final ResourceLocation location)
+    {
+        TagsProvider.TagAppender<T> builder = self();
+        builder.getInternalBuilder().removeElement(location, builder.getModID());
+        return builder;
+    }
+    
+    /**
+     * Adds multiple elements' IDs to the tag json's remove list. Callable during datageneration.
+     * @param locations The IDs of the elements to remove
+     * @return The builder for chaining
+     */
+    default TagsProvider.TagAppender<T> removeElementsByID(final ResourceLocation... locations)
+    {
+        for (ResourceLocation location : locations)
+        {
+            this.removeElementByID(location);
+        }
+        return self();
+    }
+    
+    /**
+     * Adds a tag to the tag json's remove list. Callable during datageneration.
+     * @param tag The ID of the tag to remove
+     * @return The builder for chaining
+     */
+    default TagsProvider.TagAppender<T> removeTag(Tag.Named<T> tag)
```

Comment:
Again, all of these should just be 'remove'
The only reason you need two functions {or the name changes} is because of the conflict with removeRegistryEntry
However:
`remove(T, T...)` should compile to `remove(Object, Object[])`
and
`remove(Tag.Named, Tag.Named...)` should compile to `remove(Tag.Named, Tag.Named[])` saving the type issues.
And still be able to be called with a single argument.
So you can simplify all the methods to just 'remove'.